File: RisingCodersChallenge_v1/B_IsItACM.py
t = int(input())
for testcase in range(t):
    s = input()
    A, C, M = 0, 0, 0
    for i in range(len(s)):
        if s[i] == 'A':
            A += 1
        if s[i] == 'C':
            C += 1
        if s[i] == 'M':
            M += 1

    if A == 1 and C == 1 and M == 1:
        print("ACM")
    else:
        print("NOPE")

    if 'A' in s and 'C' in s and 'M' in s:
        print("ACM")
    else:
        print("NOPE")

File: RisingCodersChallenge/RisingCodersChallenge_v1/B_IsItACM.cpp
#include <bits/stdc++.h>
using namespace std;
#define ll long long int
#define endl '\n'

void solve()
{
    set<string> s;
    s.insert("ACM");
    s.insert("AMC");
    s.insert("CAM");
    s.insert("CMA");
    s.insert("MCA");
    s.insert("MAC");
    string st;
    cin >> st;
    if (s.find(st) != s.end())
    {
        cout << "ACM\n";
        return;
    }
    cout << "NOPE\n";
}
int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int t = 1;
    cin >> t;
    while (t--)
        solve();
    return 0;
}